packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt

```diff
@@ -256,112 +281,88 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
             isValid = false
         }
 
-        if (!isValid) {
-            Datadog._internal._telemetry.debug(
+        if (isValid) {
+            rumPlugin.telemetryOverrides = telemetryOverrides
+        } else {
+            Datadog._internalProxy()._telemetry.debug(
                 String.format(MESSAGE_BAD_TELEMETRY_CONFIG, option, value)
             )
         }
     }
 
-    private fun mapTelemetryConfiguration(
-        event: TelemetryConfigurationEvent
-    ): TelemetryConfigurationEvent {
-        event.telemetry.configuration.trackViewsManually = telemetryOverrides.trackViewsManually
-        event.telemetry.configuration.trackInteractions = telemetryOverrides.trackInteractions
-        event.telemetry.configuration.trackErrors = telemetryOverrides.trackErrors
-        event.telemetry.configuration.trackNetworkRequests = telemetryOverrides.trackNetworkRequests
-        event.telemetry.configuration.trackNativeViews = telemetryOverrides.trackNativeViews
-        event.telemetry.configuration.trackCrossPlatformLongTasks =
-            telemetryOverrides.trackCrossPlatformLongTasks
-        event.telemetry.configuration.trackFlutterPerformance =
-            telemetryOverrides.trackFlutterPerformance
-        event.telemetry.configuration.dartVersion = telemetryOverrides.dartVersion
-
-        return event
-    }
-
     private fun simpleInvokeOn(methodName: String, target: Any) {
         val klass = target.javaClass
         val method = klass.declaredMethods.firstOrNull {
-            it.name == methodName || it.name == "$methodName\$dd_sdk_android_release"
+            it.name == methodName || it.name == "$methodName\$dd_sdk_android_core_release"
```

Comment:
we never had any issues with clients who are obfuscating their code?

Reply:
Only really used during flush for unit testing, so not a public feature.